invenio_app_rdm/records_ui/views/deposits.py

```diff
@@ -524,6 +524,15 @@ def deposit_edit(pid_value, draft=None, draft_files=None, files_locked=True):
             ]
         ),
     )
+    # Wrap the HTML string in a response object
+    response = make_response(html_content)
+
+    # Add cache-control headers to the response
+    response.headers["Cache-Control"] = "no-cache, no-store, must-revalidate"
+    response.headers["Pragma"] = "no-cache"
+    response.headers["Expires"] = "0"
```

Comment:
nit: Flask responses have [some fancy APIs](https://werkzeug.palletsprojects.com/en/stable/datastructures/#werkzeug.datastructures.ResponseCacheControl), that allow you to do something like:

```suggestion
    response.cache_control.no_cache = True
    response.cache_control.no_store = True
    response.cache_control.must_revalidate = True
    
    # Not 100% sure about these two
    response.pragma = "no-cache"  
    response.expires = "0"
```